### PR TITLE
Add 'Import from Postgres Dump' instructions

### LINKED ISSUE #3280: Document PGDUMP support
IMPORT will soon have support for PGDUMP output files, thanks to [cockroach#26740](https://github.com/cockroachdb/cockroach/pull/26740).

We should write some docs for it!

### PR DESCRIPTION
**Pull request by rmloveland**
- Fixes #3280

- Starts down the path to #3273

- A small piece of #3140

### CODE REVIEW

File: v2.1/import-data.md [hunk 1]
```diff
@@ -29,53 +29,112 @@ $ cockroach sql --database=[database name] < statements.sql
...
+The following options are required:
 
-    {% include copy-clipboard.html %}
-    ~~~ shell
-    $ pg_dump -t [table] [table's schema] > [filename].sql
-    ~~~
+- The `--inserts` option is required because the Postgres [`COPY`](https://www.postgresql.org/docs/current/static/sql-copy.html) statement is not yet supported
 
-For more details, see PostgreSQL's documentation on [`pg_dump`](https://www.postgresql.org/docs/9.1/static/app-pgdump.html).
+- The `--no-owner` option is required until we develop support for Postgres' object ownership statements
```
Comment (mjibson) [inline v2.1/import-data.md]: Also need the `--no-privileges` option.

Comment (rmloveland) [reply v2.1/import-data.md]: Fixed, thanks!

File: v2.1/import-data.md [hunk 2]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
-    ~~~
+- The `--no-owner` option is required until we develop support for Postgres' object ownership statements.
 
-For more details, see PostgreSQL's documentation on [`pg_dump`](https://www.postgresql.org/docs/9.1/static/app-pgdump.html).
+- The `--no-privileges` option is required until we develop support for Postgres' access privilege statements.
 
-### Reformat SQL file
+{% include copy-clipboard.html %}
+~~~ shell
+$ pg_dump --inserts -t employees --no-owner employees > /tmp/employees.sql
```
Comment (mjibson) [inline v2.1/import-data.md]: `--no-privileges` missing here

Comment (rmloveland) [reply v2.1/import-data.md]: Ach! thanks for the catch, should be fixed now.

File: v2.1/import-data.md [hunk 3]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
 
-### Create PostgreSQL SQL file
+- [Step 1. Dump the Postgres database](#step-1-dump-the-postgres-database)
+- [Step 2. `IMPORT` the Postgres dump file](#step-2-import-the-postgres-dump-file)
 
-Which `pg_dump` command you want to use depends on whether you want to import your entire database or only specific tables:
+### Step 1. Dump the Postgres database
 
-- Entire database:
+To dump the 'employees' table from a Postgres database also named 'employees', run the [`pg_dump`](https://www.postgresql.org/docs/current/static/app-pgdump.html) command below.
```
Comment (Amruta-Ranade) [inline v2.1/import-data.md]: 'employees' > `employees`

File: v2.1/import-data.md [hunk 4]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
 
-    {% include copy-clipboard.html %}
-    ~~~ shell
-    $ pg_dump [database] > [filename].sql
-    ~~~
+The following options are required:
```
Comment (Amruta-Ranade) [inline v2.1/import-data.md]: Suggestion: Convert the bulleted list into a table and place it after the command

File: v2.1/import-data.md [hunk 5]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
 
-After generating the `.sql` file, you need to perform a few editing steps before importing it:
+### Step 2. `IMPORT` the Postgres dump file
 
-1. Remove all statements from the file besides the `CREATE TABLE` and `COPY` statements.
-2. Manually add the table's [`PRIMARY KEY`](primary-key.html#syntax) constraint to the `CREATE TABLE` statement.
-  This has to be done manually because PostgreSQL attempts to add the primary key after creating the table, but CockroachDB requires the primary key be defined upon table creation.
-3. Review any other [constraints](constraints.html) to ensure they're properly listed on the table.
-4. Remove any [unsupported elements](sql-feature-support.html).
+To import the Postgres dump file, issue an `IMPORT` statement like the one below.
```
Comment (Amruta-Ranade) [inline v2.1/import-data.md]: This sentence is disconnected from the paragraph that follows it, and the `IMPORT` statement it's referring to is much later in the doc. Can we start this para with "You will need..." and add this sentence right before the `IMPORT` statement?

File: v2.1/import-data.md [hunk 6]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
+~~~
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+|       job_id       |  status   | fraction_completed | rows | index_entries | system_records | bytes |
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+| 352938237301293057 | succeeded |                  1 |    0 |             0 |              0 |     0 |
++--------------------+-----------+--------------------+------+---------------+----------------+-------+
+(1 row)
+~~~
+
+Finally, we can verify that despite needing to edit the `gender` column's data type in the Postgres dump file, the column was populated correctly during the import:
```
Comment (Amruta-Ranade) [inline v2.1/import-data.md]: This sentence makes me think that we were expecting something to go wrong and are glad it didn't :) 
Also, I think we would want to verify the whole table was imported correctly. Would rewording the sentence to reflect that be better?

File: v2.1/import-data.md [hunk 7]
```diff
@@ -29,53 +29,114 @@ $ cockroach sql --database=[database name] < statements.sql
...
+To dump the 'employees' table from a Postgres database also named 'employees', run the [`pg_dump`](https://www.postgresql.org/docs/current/static/app-pgdump.html) command below.
 
-    {% include copy-clipboard.html %}
-    ~~~ shell
-    $ pg_dump [database] > [filename].sql
-    ~~~
+The following options are required:
 
-- Specific tables:
+- The `--inserts` option is required because the Postgres [`COPY`](https://www.postgresql.org/docs/current/static/sql-copy.html) statement is not yet supported.
```
Comment (Amruta-Ranade) [inline v2.1/import-data.md]: Suggestion: "The `--inserts` option is required until we develop ...." to maintain parallelism with the other options.

Comment (rmloveland) [reply v2.1/import-data.md]: Fixed.

File: v2.1/import-data.md (hunk 6 above)
Comment (rmloveland) [reply v2.1/import-data.md]: > This sentence makes me think that we were expecting something to go wrong and are glad it didn't :)

I was expecting an issue after having to edit the Postgres dump file, and was pleasantly surprised when CockroachDB magically imported it correctly. :-)  I agree it's a bit strange.  I imagine most people would want to verify that a data type they hand-edited (which implies that it might not work) was imported correctly.  This is an area I suspect will change as `IMPORT` continues to add support for Postgres features.

> Also, I think we would want to verify the whole table was imported correctly. Would rewording the sentence to reflect that be better?

I'm not sure how to document this in the general case.  I think this will probably be specific to the user's application.  I imagine they would need to run their application and various tests under load against CRDB to really confirm that it works as expected for them.

File: v2.1/import-data.md (hunk 5 above)
Comment (rmloveland) [reply v2.1/import-data.md]: Fixed.

File: v2.1/import-data.md (hunk 4 above)
Comment (rmloveland) [reply v2.1/import-data.md]: Fixed.

File: v2.1/import-data.md (hunk 3 above)
Comment (rmloveland) [reply v2.1/import-data.md]: Fixed.

**Comment by jseldess** [comment]
:lgtm:, with some nits.

---

Review status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

---

*[v2.1/import-data.md, line 39 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQP890EkUQXhW-dE7P:-LGQP890EkUQXhW-dE7Q:b-wo1fov) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L39)):*
> ```Markdown
> - [Step 2. `IMPORT` the Postgres dump file](#step-2-import-the-postgres-dump-file)
> 
> ### Step 1. Dump the Postgres database
> ```

nit: `Postgres` > `PostgreSQL` here and wherever else it shows up on this page. 

---

*[v2.1/import-data.md, line 52 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQPTMf0qZ03kukm8qh:-LGQPTMf0qZ03kukm8qi:b-uaymrp) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L52)):*
> ```Markdown
>  Option            | Description
> -------------------+--------------------------------------------------------------------------------------------------------------------------------------
>  `--inserts`       | Required until we develop support for the Postgres [`COPY`](https://www.postgresql.org/docs/current/static/sql-copy.html) statement.
> ```

nit: `Required until we develop support for...` > `Required until CockroachDB implements support for...`

Please apply below as well.

---

*[v2.1/import-data.md, line 95 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQQBbdBNvSLaxVt_SP:-LGQQBbdBNvSLaxVt_SQ:bvstzt9) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L95)):*
> ```Markdown
> ~~~
> 
> Finally, we can confirm that despite needing to edit the `gender` column's data type in the Postgres dump file, the column was populated correctly during the import:
> ```

Let's just remove `we` here: `Finally, confirm that despite needing to edit..`

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3306#-:-LGQQPlA81VIkdgdzEo6:brn2xfh)*

**Comment by rmloveland** [comment]
Review status: :shipit: complete! 1 of 0 LGTMs obtained

---

*[v2.1/import-data.md, line 39 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQP890EkUQXhW-dE7P:-LGQSdS7BJu_FmqXLLzf:baz1pkq) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L39)):*
Fixed.  I think we should consider allowing "Postgres" in future though, since according to [the PG docs](https://www.postgresql.org/docs/10/static/history.html)

> Many people continue to refer to PostgreSQL as “Postgres” (now rarely in all capital letters) because of tradition or because it is easier to pronounce. This usage is widely accepted as a nickname or alias.

Perhaps something to discuss for style guide with @lhirata

---

*[v2.1/import-data.md, line 52 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQPTMf0qZ03kukm8qh:-LGQXTKi7obyQlt_cl4i:bt2vr1b) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L52)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `Required until we develop support for...` > `Required until CockroachDB implements support for...`

Please apply below as well.

</blockquote></details>

Fixed.

---

*[v2.1/import-data.md, line 95 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQQBbdBNvSLaxVt_SP:-LGQXVP3COxNemCXkwzl:bt2vr1b) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L95)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Let's just remove `we` here: `Finally, confirm that despite needing to edit..`
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3306)*

**Comment by jseldess** [comment]
*[v2.1/import-data.md, line 39 at r4](https://reviewable.io/reviews/cockroachdb/docs/3306#-LGQP890EkUQXhW-dE7P:-LGQdgxY4axlIrfaIKr6:bgeklqw) ([raw file](https://github.com/cockroachdb/docs/blob/9bbd43db7ec8057171bc3ed27d88928bfac245a8/v2.1/import-data.md#L39)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Fixed.  I think we should consider allowing "Postgres" in future though, since according to [the PG docs](https://www.postgresql.org/docs/10/static/history.html)

> Many people continue to refer to PostgreSQL as “Postgres” (now rarely in all capital letters) because of tradition or because it is easier to pronounce. This usage is widely accepted as a nickname or alias.

Perhaps something to discuss for style guide with @lhirata
</blockquote></details>

Yeah, I'm probably too strict here and am glad to loosen this. Good idea to consider it for style guide.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3306)*
